statistics/table.go

```diff
@@ -355,11 +379,88 @@ func (t *Table) GetRowCountByIndexRanges(sc *stmtctx.StatementContext, idxID int
 		}
 		return getPseudoRowCountByIndexRanges(sc, indexRanges, float64(t.Count), colsLen)
 	}
-	result, err := idx.getRowCount(sc, indexRanges)
+	var result float64
+	var err error
+	if idx.CMSketch != nil && idx.statsVer == version1 {
+		result, err = t.getIndexRowCount(sc, idx, indexRanges)
+	} else {
+		result, err = idx.getRowCount(sc, indexRanges)
+	}
 	result *= idx.getIncreaseFactor(t.Count)
 	return result, errors.Trace(err)
 }
 
+func getRangePosition(sc *stmtctx.StatementContext, ran *ranger.Range) int {
```

Comment:
1. maybe `getOrdinalOfRangeCond` is a better name?
2. add some comment for this function